### PR TITLE
Fixed #318 upload multiple images from gallery

### LINKED ISSUE #318: Selected Images in main gallery cannot be uploaded to upload list
**Actual Behaviour**

If we select images in main screen multiple images cannot be uploaded to upload list.

**Expected Behaviour**

There should be an upload button also with delete option when you select multiple images.

**Steps to reproduce it**

Add steps to reproduce bugs or add information on the place where the feature should be implemented. Add links to a sample deployment or code.

**LogCat for the issue**

Provide logs for the crash here

**Screenshots of the issue**

17820041_781557941995315_906642191_o

**Would you like to work on the issue?**

I have assigned myself an other issue so anyone willing can work on it.

### PR DESCRIPTION
**Pull request by mohitmanuja**
#318 

Changes: Now user can upload multiple images from gallery fragment. 
- added upload option in the menu.

Screenshots for the change: 

screenshot_2017-04-07-22-47-29-574_vn mbm phimp me
screenshot_2017-04-07-22-47-32-923_vn mbm phimp me

### CODE REVIEW

**Comment by phimpme-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">:tada: Thanks for the PR @mohitmanuja we hope you have changed the variables in the upload_apk_forks.sh its for easy moderation.</td>
    </tr>
  </tbody>
</table>

:link: Maintainers: the apk to test can be found at https://github.com/mohitmanuja/phimpme-android/blob/apk/app-debug.apk

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: app/src/main/java/vn/mbm/phimp/me/newGallery.java [hunk 1]
```diff
@@ -529,6 +532,12 @@ public void onClick(DialogInterface dialog, int id) {
                 getActivity().setTitle(newTitle);
                 getActivity().invalidateOptionsMenu();
                 return true;
+            case R.id.menu_gallery_upload:
+                for(int i=0;i<deletableList.size();i++){
+                    Upload.uploadGridList.add(deletableList.get(i));
```
Comment (pa1pal) [inline app/src/main/java/vn/mbm/phimp/me/newGallery.java]: Check for the duplication of the image like implemented in `Upload.java`.

File: app/src/main/java/vn/mbm/phimp/me/newGallery.java [hunk 2]
```diff
@@ -529,6 +532,12 @@ public void onClick(DialogInterface dialog, int id) {
...
+                }
+                Toast.makeText(ctx, deletableList.size()+ " Images added to upload list.", Toast.LENGTH_SHORT).show();
```
Comment (pa1pal) [inline app/src/main/java/vn/mbm/phimp/me/newGallery.java]: It will be "images" for one image as well. Add a check here.

Comment (mohitmanuja) [inline app/src/main/java/vn/mbm/phimp/me/newGallery.java]: Sure

**Comment by mariobehling** [comment]
@mohitmanuja Please help to resolve the outstanding points. Thanks.

**Comment by mohitmanuja** [comment]
Thanks for pointing out. I have done the changes. Please review it.

File: app/src/main/java/vn/mbm/phimp/me/newGallery.java [hunk 3]
```diff
@@ -536,9 +536,17 @@ public void onClick(DialogInterface dialog, int id) {
...
+                        Upload.uploadGridList.add(deletableList.get(i));
+                    }
                 }
-                Toast.makeText(ctx, deletableList.size()+ " Images added to upload list.", Toast.LENGTH_SHORT).show();
+                if(deletableList.size()>1)
+                    Toast.makeText(ctx, deletableList.size()+" Images added to upload list.", Toast.LENGTH_SHORT).show();
+                    else
+                    Toast.makeText(ctx, deletableList.size()+" Image added to upload list.", Toast.LENGTH_SHORT).show();
+
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/newGallery.java]: Unwanted blank lines?

**Pull request merged**